server/impl/src/main/java/com/walmartlabs/concord/server/user/UserResourceV2.java

```diff
@@ -66,12 +69,32 @@ public List<UserEntry> list(@QueryParam("offset") @DefaultValue("0") int offset,
         return userDao.list(filter, offset, limit);
     }
 
+
+    /*
+        throws unauthorized exception if user is not an admin,
+        copied from UserResource.java
+    */
+    private static void assertAdmin() {
+        if (!Roles.isAdmin()) {
+            throw new UnauthorizedException("Only admins can do that");
+        }
+    }
+
     @GET
     @Path("/{id}")
     @Produces(MediaType.APPLICATION_JSON)
     @WithTimer
     @Operation(description = "Get an existing user", operationId = "getUser")
     public UserEntry get(@PathParam("id") UUID id) {
+
+        UserPrincipal loggedIn = UserPrincipal.assertCurrent();
+
+        UUID authenticatedId = loggedIn.getId();
+
+        if(!authenticatedId.equals(id)) {
+            assertAdmin();
+        }
```

Comment:
Let's inline the assertAdmin method and just do `...if (!Roles.isAdmin())...` right here. And let's be more specific in the error message. In the old UserResource the assert was strict all or nothing, and here it is more complicated.

```suggestion
        if(!authenticatedId.equals(id) && !Roles.isAdmin()) {
            throw new UnauthorizedException("Users can only view their own information or must have admin privileges.");
        }
```

or something like that.